rest-api-sdk/src/main/java/com/paypal/base/SSLUtil.java

```
@@ -292,20 +293,24 @@ public static InputStream downloadCertificateFromPath(String urlPath) throws Pay
 	 * Generates a CRC 32 Value of String passed
 	 * 
 	 * @param data
-	 * @return long crc32 value of input. -1 if string is null
+	 * @return long crc32 value of input. -1 if string is null or UTF-8 is not a valid character encoding
 	 */
 	public static long crc32(String data) {
 		if (data == null) { 
 			return -1;
 		}
 
-		// get bytes from string
-		byte bytes[] = data.getBytes();
-		Checksum checksum = new CRC32();
-		// update the current checksum with the specified array of bytes
-		checksum.update(bytes, 0, bytes.length);
-		// get the current checksum value
-		return checksum.getValue();
+		try {
+			// get bytes from string
+			byte bytes[] = data.getBytes("UTF-8");
+			Checksum checksum = new CRC32();
+			// update the current checksum with the specified array of bytes
+			checksum.update(bytes, 0, bytes.length);
+			// get the current checksum value
+			return checksum.getValue();
+		} catch (UnsupportedEncodingException e) {
+			return -1;
```

I agree too. If that is the case, can we just throw the UnsupportedEncodingException ?